### PR TITLE
Restructurized BNF, and comprehension tests

### PR DESCRIPTION
**Pull request by lkadalski**
Upgraded BNF with tests. 
In meantime I found out that many of tests cases are "wrong". (using not existing variables and forbidden keywords)
Surely they need some rework.

### CODE REVIEW

File: src/main/grammar/RegoLexer.flex [hunk 1]
```diff
@@ -73,6 +73,7 @@ VAR=[\p{Alpha}_][\p{Alnum}_]*
   "`"                   { return BACKTICK; }
   "."                   { return DOT; }
   "_"                   { return UNDER; }
+  "term"                { return TERM; }
```
Comment (lukasz-kaminski) [inline src/main/grammar/RegoLexer.flex]: Wait, it's a keyword now?

File: src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego [hunk 2]
```diff
@@ -9,10 +9,3 @@ app_to_hostnames := {app.name: hostnames |
                     s.name == name
                     hostname := s.hostname]
 }
-
-
-
-merge_objects(a, b) = c {
```
Comment (lukasz-kaminski) [inline src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego]: Why did you delete this case?

Comment (lkadalski) [reply src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego]: This is not an object comprehension, it's a set comprehension. So a different case.

File: src/main/grammar/RegoLexer.flex (hunk 1 above)
Comment (lkadalski) [reply src/main/grammar/RegoLexer.flex]: ehm....no ;) lemme delete this

File: src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego (hunk 2 above)
Comment (lukasz-kaminski) [reply src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego]: Shouldn't we move it to appropriate file then?

Comment (lkadalski) [reply src/test/resources/org/opa/ideaplugin/lang/parser/fixtures/object_comprehension.rego]: I moved it to a separate test case

**Review by lukasz-kaminski — APPROVED** [review]
